examples/dataflow-streaming-benchmark/src/main/java/com/google/cloud/pso/pipeline/StreamingBenchmark.java

```diff
@@ -194,24 +260,31 @@ public void setup() throws IOException {
         }
 
         schema = byteArrayOutputStream.toString();
+        this.validateSchema(dataGenerator, schema);
       }
     }
 
     @ProcessElement
     public void processElement(ProcessContext context)
         throws IOException, JsonDataGeneratorException {
 
-      // TODO: Add the ability to place eventId and eventTimestamp in the attributes.
       byte[] payload;
       Map<String, String> attributes = Maps.newHashMap();
-
       // Generate the fake JSON according to the schema.
       try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
         dataGenerator.generateTestDataJson(schema, byteArrayOutputStream);
-
         payload = byteArrayOutputStream.toByteArray();
       }
 
+      if (this.includeAttributeValues) {
+        // Build a json Object, extract fields and include them in attributes
+        JsonObject jsonObject =
+            this.jsonParser.parse(new String(payload, StandardCharsets.UTF_8)).getAsJsonObject();
+        for (String field : attributeFields) {
+          attributes.put(field, jsonObject.get(field).getAsString());
+        }
```

Comment:
This step of deserializing the payload to pull out the attributes seems like it will create some overhead. Have you checked if the guidance in the README of ~2,500 QPS per core still remains valid in the case that we are publishing with attributes.

Reply:
Yes validated and performance overhead is very small.  Here are the benchmarks computed as average of 3 iterations:

1000 Messages - jsonOnly(293ms), json + DeSerialization(303ms)
2500 Messages - jsonOnly(494ms), json + DeSerialization(544ms)
5000 Messages - jsonOnly(796ms), json + DeSerialization(855ms)
10000 Messages - jsonOnly(1105ms), json + DeSerialization(1183ms)

Reply:
LGTM. Thanks for validating!